eng/common/scripts/Generate-DocIndex.ps1

```diff
@@ -0,0 +1,149 @@
+        return 
+    }
+    return $filteredPkg[0]
+}
+
+function Generate-DocIndex { 
+    Param (
+        [Parameter(Mandatory = $true)]  [Object[]]  $metadata,
+        [Parameter(Mandatory = $true)]  [Hashtable] $filterOptions,
+        [Parameter(Mandatory = $true)]  [String]    $blobStorageUrl,
+        [Parameter(Mandatory = $true)]  [String]    $blobDirectoryRegex,
+        [Parameter(Mandatory = $true)]  [String]    $titleRegex,
+        [Parameter(Mandatory = $false)] [String]    $packageNameRegex = '',
+        [Parameter(Mandatory = $false)] [String]    $packageNameReplacement = ''
+    )
+    Write-Verbose "Name Reccuring paths with variable names"
+    $DocFxTool = "${RepoRoot}/docfx/docfx.exe"
+    $DocOutDir = "${RepoRoot}/docfx_project"
+
+    Write-Verbose "Initializing Default DocFx Site..."
+    & $($DocFx) init -q -o "${DocOutDir}"
+
+    Write-Verbose "Copying template and configuration..."
+    New-Item -Path "${DocOutDir}" -Name "templates" -ItemType "directory" -Force
+    Copy-Item "${DocGenDir}/templates/*" -Destination "${DocOutDir}/templates" -Force -Recurse
+    Copy-Item "${DocGenDir}/docfx.json" -Destination "${DocOutDir}/" -Force
+    $YmlPath = "${DocOutDir}/api"
+    New-Item -Path $YmlPath -Name "toc.yml" -Force
+
+    Write-Verbose "Reading artifact from storage blob ..."
+    $pageToken = ""
+    # Used for sorting the toc display order
+    $orderSet = @{}
+    # This is a pagnation call as storage only return 5000 results as maximum.
+    Do {
+        $resp = ""
+        if (!$pageToken) {
+            # First page call.
+            $resp = Invoke-RestMethod -Method Get -Uri $blobStorageUrl
+        }
+        else {
+            # Next page call
+            $blobStorageUrlPageToken = $blobStorageUrl + "&marker=$pageToken"
+            $resp = Invoke-RestMethod -Method Get -Uri $blobStorageUrlPageToken
+        }
+        # Storage returns some weired encoded string at the begining of response which needs to cutoff before use.
+
+        # Convert to xml documents.
+        $xmlDoc = [xml](removeBomFromString $resp)
+        foreach ($elem in $xmlDoc.EnumerationResults.Blobs.BlobPrefix) {
+            # What service return like "dotnet/Azure.AI.Anomalydetector/", needs to fetch out "Azure.AI.Anomalydetector"
+            $artifact = $elem.Name -replace $blobDirectoryRegex, '$1'
+            # Some languages need to convert the artifact name, e.g azure-data-appconfiguration -> @azure/data-appconfiguration
+            if ($packageNameRegex) {
+                $artifact = $artifact -replace $packageNameRegex, $packageNameReplacement
+            }
+            $packageInfo = Get-PkgMetadata -metadata $metadata -artifactName $artifact -filterOption $filterOptions
+            $serviceName = ""
+            if (!$packageInfo) {
+                LogWarning "Did not find the artifacts $artifact from release csv. Please check and update."
```

Comment:
We should also set service name to "Other" in this case when it isn't found in the metadata.

Reply:
Yes, it is inside of elseif block

Reply:
I assume that packageInfo would be null in the case where there is no package found which is part of the first if block.

Reply:
If no such artifact in csv, we don't do any updates for the package, so there is no need to update service name as well. But if artifact exists, but no service name, we can put it to `Other`. 

What's the purpose of setting service name to other if no such artifact in csv?

Reply:
Consider the case where we have just released a new library. We want it to show up in to the doc index but it will not have yet been added to the csv file yet until that process kicks off. I want to make sure it shows up in the index even if it isn't yet in the csv file.